backend/server.js

```diff
@@ -3,10 +3,39 @@ const cors = require("cors");
 const app = express();
 const port = 3000;
 
+const admin = require("firebase-admin");
+const serviceAccount = require("./serviceAccountKey.json");
+const { getAuth } = require("firebase-admin/auth");
 app.use(cors());
+app.use(express.json());
 const sqlite3 = require("sqlite3").verbose();
 const db = new sqlite3.Database("db.sqlite3");
 
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccount),
+});
+
+async function addUser(email, password) {
+  try {
+    const userRecord = await getAuth().createUser({
+      email: email,
+      password: password,
+    });
+    console.log("Successfully created new user:", userRecord.uid);
+    return userRecord;
+  } catch (error) {
+    console.error("Error creating new user:", error);
+    throw error;
+  }
+}
+app.post("/add-user", (req, res) => {
+  const email = req.body.email;
+  const password = req.body.password;
+  addUser(email, password).then(() => {
+    res.status(201).json({ message: "Käyttäjä luotu!" });
+  });
```

Comment:
Can we add basic error handling here by adding a catch() after then() and send a response to the frontend with appropriate error status code and message?